src/js/controllers/navbar.js

```diff
@@ -1,6 +1,29 @@
 'use strict';
 module.exports = function(app) {
-  app.controller('NavbarCtrl', ['$scope', function($scope) {
-    $scope.collapsed = true;
-  }]);
+  app.controller(
+    'NavbarCtrl',
+    ['$scope', '$http', '$location', '$window', '$routeSegment', 'config',
+      function($scope, $http, $location, $window, $routeSegment, config) {
+        $scope.collapsed = true;
+
+        $scope.search = {};
+        $scope.phSearch = function(query, limit) {
+          return $http.get(config.apiUrl + '/articles/', {
+            params: {q: query, limit: limit}
+          })
+          .then(function(response) {
+            return response.data;
+          });
+        };
+
+        $scope.goToArticle = function(item, model, label) {
+          $location.path($routeSegment.getSegmentUrl(
+            'articles', {articleId: item._id}
+          ));
+          // for some reason, $route.reload() isn't enough here
+          $window.location.reload();
```

Comment:
it should even work without this line, see [here](https://github.com/paperhub/paperhub-frontend/blob/master/src/js/controllers/articleNew.js#L17)